src/Microsoft.ML.TorchSharp/Utils/FileUtils.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+
+namespace Microsoft.ML.TorchSharp.Utils
+{
+    internal static class FileUtils
+    {
+        private const int WriteBatchSize = 1024 * 1024;     // this size seems good on performance
+        private static readonly Type[] _validTypes = new Type[]
+        {
+            typeof(short),
+            typeof(ushort),
+            typeof(int),
+            typeof(uint),
+            typeof(long),
+            typeof(ulong),
+            typeof(float),
+            typeof(double),
+        };
+
+        public static string LoadFromFileOrDownloadFromWeb(string path, string fileName, Uri url)
```

Comment:
>path [](http://example.com/codeflow?start=66&length=4)

do we every can pass null here? or do we want to support null if someone want to call this method for using Url only?

Reply:
never mind, I am seeing we can safe the web downloaded file there too. Now I suggest adding assert (path != null && fileName != null)

Reply:
Added those checks in.